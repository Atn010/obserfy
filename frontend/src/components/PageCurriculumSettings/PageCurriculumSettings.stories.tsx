import React, { FC } from "react"
import PageCurriculumSettings from "./PageCurriculumSettings"

export default {
  title: "Core|PageCurriculumSettings",
  component: PageCurriculumSettings,
  parameters: {
    componentSubtitle: "Just a simple PageCurriculumSettings",
  },
}

export const Basic: FC = () => <PageCurriculumSettings />
