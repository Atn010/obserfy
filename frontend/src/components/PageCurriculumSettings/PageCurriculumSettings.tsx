import React, { FC } from "react"
import Typography from "../Typography/Typography"
import { Box } from "../Box/Box"
import Flex from "../Flex/Flex"
import Spacer from "../Spacer/Spacer"
import Button from "../Button/Button"
import { getAnalytics } from "../../analytics"
import { getSchoolId } from "../../hooks/schoolIdState"
import useApi from "../../hooks/useApi"

export const PageCurriculumSettings: FC = () => {
  const [curriculum, setCurriculumOutdated] = useApi(
    `/schools/${getSchoolId()}/curriculum`
  )

  async function createNewDefaultCurriculum(): Promise<void> {
    const baseUrl = "/api/v1"
    const response = await fetch(
      `${baseUrl}/schools/${getSchoolId()}/curriculum`,
      {
        credentials: "same-origin",
        method: "POST",
      }
    )
    setCurriculumOutdated()
    getAnalytics()?.track("Default curriculum created", {
      responseStatus: response.status,
    })
  }
  return (
    <Box maxWidth="maxWidth.sm" margin="auto">
      {!curriculum && (
        <Flex alignItems="center" p={3}>
          <Typography.H6>Setup curriculum</Typography.H6>
          <Spacer />
          <Button onClick={createNewDefaultCurriculum}>Use default</Button>
        </Flex>
      )}
    </Box>
  )
}

export default PageCurriculumSettings
