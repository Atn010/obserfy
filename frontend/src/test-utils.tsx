/* eslint-disable import/no-extraneous-dependencies */
import React, { FC, ReactElement } from "react"
import { ThemeProvider } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import { IntlProvider } from "gatsby-plugin-intl3"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import Theme from "./gatsby-plugin-theme-ui"
import Box from "./components/Box/Box"
import en from "./intl/en.json"

const AllProviders: FC = ({ children }) => {
  // TODO: Maybe this better be outside
  const history = createHistory(createMemorySource("/"))
  return (
    <LocationProvider history={history}>
      <ThemeProvider theme={Theme}>
        <IntlProvider locale="en" messages={en}>
          <Box fontSize={[16, 20]}>{children}</Box>
        </IntlProvider>
      </ThemeProvider>
    </LocationProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
