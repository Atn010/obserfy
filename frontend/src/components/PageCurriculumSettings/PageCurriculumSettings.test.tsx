import React from "react"
import { render } from "../../test-utils"
import { Basic } from "./PageCurriculumSettings.stories"

describe("PageCurriculumSettings", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
