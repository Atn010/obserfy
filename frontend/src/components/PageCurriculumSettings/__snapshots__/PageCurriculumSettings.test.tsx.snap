// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageCurriculumSettings should render correctly 1`] = `
@keyframes animation-0 {
  0% {
    background-color: rgba(165,165,165,0.05);
  }

  50% {
    background-color: rgba(165,165,165,0.1);
  }

  100% {
    background-color: rgba(165,165,165,0.05);
  }
}

@keyframes animation-0 {
  0% {
    background-color: rgba(165,165,165,0.05);
  }

  50% {
    background-color: rgba(165,165,165,0.1);
  }

  100% {
    background-color: rgba(165,165,165,0.05);
  }
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width:40em) {
  .emotion-6 {
    font-size: 20px;
  }
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 640px;
  margin: auto;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 6px;
  -webkit-animation: 1s ease-in-out 0s infinite animation-0;
  animation: 1s ease-in-out 0s infinite animation-0;
  width: 100%;
  height: 5rem;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 6px;
  -webkit-animation: 1s ease-in-out 0s infinite animation-0;
  animation: 1s ease-in-out 0s infinite animation-0;
  width: 100%;
  height: 6rem;
  margin-top: 16px;
}

<div>
  <div
    class="emotion-6"
  >
    <div
      class="emotion-5"
    >
      <div
        class="emotion-4"
      >
        <div
          class="emotion-0"
        />
        <div
          class="emotion-1"
        />
        <div
          class="emotion-1"
        />
        <div
          class="emotion-1"
        />
      </div>
    </div>
  </div>
</div>
`;
